getgather/distill.py

```diff
@@ -297,14 +298,24 @@ async def distill(hostname: str | None, page: Page, patterns: list[Pattern]) ->
         return match
 
 
-async def run_distillation_loop(location: str, patterns: list[Pattern], fields: list[str] = []):
+async def run_distillation_loop(
+    location: str,
+    patterns: list[Pattern],
+    fields: list[str] = [],
+    browser_profile: BrowserProfile | None = None,
+):
     if len(patterns) == 0:
         logger.error("No distillation patterns provided")
         raise ValueError("No distillation patterns provided")
 
     hostname = urllib.parse.urlparse(location).hostname or ""
 
-    profile = BrowserProfile()
+    # Use provided profile, or try to get from MCP context, or create new one
+    if browser_profile:
+        profile = browser_profile
+    else:
+        profile = get_mcp_browser_profile() or BrowserProfile()
```

Comment:
`get_mcp_browser_profile` should be invoked by the caller (e.g. `get_book_list()`) since this distillation loop is at the layer where it does not really know about MCP.